source/ptreferential/ptreferential_utils.cpp

```diff
@@ -291,30 +291,35 @@ type::Indexes get_indexes_from_code(const type::Type_e type,
 template <typename T>
 static
     typename boost::enable_if<typename boost::mpl::contains<nt::CodeContainer::SupportedTypes, T>::type, Indexes>::type
-    get_indexes_from_code_type(const std::string& key, const Data& data) {
+    get_indexes_from_code_type(const std::vector<std::string>& keys, const Data& data) {
     Indexes indexes;
     auto collection = data.pt_data->collection<T>();
     for (const auto* obj : collection) {
         auto codes = data.pt_data->codes.get_codes<T>(obj);
-        if (codes.find(key) == codes.end()) {
-            continue;
+        for (const auto& key : keys) {
+            if (codes.find(key) == codes.end()) {
+                continue;
+            }
+            indexes.insert(obj->idx);
```

Comment:
You may also stop iterating on keys once you found one (`break` here, or reverse if condition for elegance)

Reply:
done :heavy_check_mark: